core/src/main/scala/org/apache/spark/deploy/master/Master.scala

```diff
@@ -524,49 +524,26 @@ private[master] class Master(
   }
 
   /**
-   * Can an app use the given worker? True if the worker has enough memory and we haven't already
-   * launched an executor for the app on it (right now the standalone backend doesn't like having
-   * two executors on the same worker).
+   * Can an app use the given worker?
    */
   private def canUse(app: ApplicationInfo, worker: WorkerInfo): Boolean = {
-    worker.memoryFree >= app.desc.memoryPerSlave && !worker.hasExecutor(app)
+    val enoughResources = worker.memoryFree >= app.desc.memoryPerExecutorMB && worker.coresFree > 0
+    val allowToExecute = app.desc.coresPerExecutor.isDefined || !worker.hasExecutor(app)
```

Comment:
I don't think we need this check. This should already be handled downstream because `worker.coresFree` will be 0 after we launch the first executor. Can you update the javadocs to say:
```
Return whether the worker has enough resources for the app to launch executors on it.
```